libcodechecker/server/api/report_server.py

```diff
@@ -394,28 +407,49 @@ def has_ongoing_run(self, run_id):
             return run_id in self.__sessions
 
         def get_transaction(self, run_id):
-            self.__sessions[run_id]['timer'] = time.time()
-            return self.__sessions[run_id]['transaction']
-
-        def _timeout_sessions(self):
+            """
+            Retrieve the SQLAlchemy session that encapsulated the transaction
+            for the given run_id's store operation.
+            """
+            session = self.__sessions[run_id]
+            session['timer'] = time.time()
+
+            # Because the run was "accessed" by a store operation, we update
+            # the run's timestamp.
+            if int(time.time() - session['last_lock_update']) > 5 * 60:
+                # For easing the database from writes, only update the lock
+                # every five minutes. This change must be committed to the
+                # database with no regards to the undergoing operations.
+                lock_update_session = self.__session_factory()
+                run = lock_update_session.query(Run).get(run_id)
```

Comment:
I think we can use `update` on the session object here:
```py
lock_update_session.query(Run)
    .filter(Run.id == run_id)
    .update({Run.lock_timestamp : datetime.now()})
```
For more info see: http://docs.sqlalchemy.org/en/latest/orm/session_basics.html